File: backend/src/main/java/com/exivamoeres/service/AuthService.java
package com.exivamoeres.service;

import com.exivamoeres.dto.auth.AuthResponse;
import com.exivamoeres.dto.auth.LoginRequest;
import com.exivamoeres.dto.auth.RegisterRequest;

/** Autenticação por email/senha e contas anônimas (OAuth fica no pacote security). */
public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    /**
     * Cria um usuário anônimo e já devolve tokens — permite usar o site sem
     * cadastro. A promoção de conta anônima para registrada está prevista
     * para a sessão 2 (ver docs/proxima-sessao.md).
     */
    AuthResponse loginAnonymous(String displayName);

    AuthResponse refresh(String refreshToken);

    void logout(String refreshToken);
}

File: frontend/src/services/authService.ts
import { apiClient } from './apiClient'
import type { AuthResponse } from '../types/api'

export const authService = {
  register: (email: string, password: string, displayName: string) =>
    apiClient
      .post<AuthResponse>('/api/auth/register', { email, password, displayName })
      .then((r) => r.data),

  login: (email: string, password: string) =>
    apiClient.post<AuthResponse>('/api/auth/login', { email, password }).then((r) => r.data),

  loginAnonymous: (displayName?: string) =>
    apiClient.post<AuthResponse>('/api/auth/anonymous', { displayName }).then((r) => r.data),

  refresh: (refreshToken: string) =>
    apiClient.post<AuthResponse>('/api/auth/refresh', { refreshToken }).then((r) => r.data),

  logout: (refreshToken: string) =>
    apiClient.post<void>('/api/auth/logout', { refreshToken }).then(() => undefined),
}
